### PR TITLE
Revolutionise the NFT ecosystem: introduce slavery

### PR DESCRIPTION
**Pull request by craigfe**
The maths checks out, so it is just. Code is law.

CC @maiste.

### CODE REVIEW

File: purse [hunk 1]
```diff
@@ -4,11 +4,12 @@ TheLortex:
   - I have 9 Gitcoins
 
 maiste:
-  - I also have 10 Gitcoins
+  - I also have 12 Gitcoins
```
Comment (maiste) [inline purse]:
```suggestion
  - I also have 10.5 Gitcoins

```

File: purse [hunk 2]
```diff
@@ -4,11 +4,12 @@ TheLortex:
...
 
 MisterDA:
   - I have drawn a beautiful camel
   - I have 11 Gitcoins
 
 CraigFe:
-  - I have 10 Gitcoins
+  - I have 8 Gitcoins
```
Comment (maiste) [inline purse]:
```suggestion
  - I have 9.5 Gitcoins
```
I'm sorry to bother you with this change, but owning @maiste user for 2 Gitcoins is an unfair price. To consider this transaction valide, you need to put a fair price: I suggest 0.5 Gitcoin

**Review by maiste — CHANGES_REQUESTED** [review]

File: purse (hunk 2 above)
Comment (craigfe) [reply purse]: I see your offer of 0.5 Gitcoins and raise you `(if P == NP then 0.5 else 2)` Gitcoins.

I suggest you accept before I manage to merge a "_Githereum balances can be computationally undecidable._" clause.

Comment (maiste) [reply purse]: I see you are hard in business @CraigFe! @TheLortex, as a wise Githereum dictator should have the last word about it.

Comment (TheLortex) [reply purse]: Interesting.
Computationally indecidable balances seems to be great addition to the blockchain. So the _math checks out_ rule is extended to _math has to check out in any possible universe_, meaning that balances have to stay positive in both the P == NP and P != NP case. Balances are collapsed when the problem is solved, using a Proof of Proof mechanism.

Comment (TheLortex) [reply purse]: @CraigFe Please take a look at https://github.com/TheLortex/githereum/pull/19 and update the PR accordingly to add the conditional balances.

File: purse [hunk 3]
```diff
@@ -4,11 +4,12 @@ TheLortex:
...
+  - I own a token that abstractly represents ownership of @maiste
```
Comment (maiste) [inline purse]:
```suggestion
  - I have 10 Gitcoins if P != NP
  - I have 8 Gitcoins if P = NP
  - I own a token that abstractly represents ownership of @maiste if P = NP
```

File: purse (hunk 1 above)
Comment (maiste) [inline purse]:
```suggestion
  - I also have 12 Gitcoins if P = NP
  - I also have 10 Gitcoins if P != NP
```

File: purse (hunk 2 above)
Comment (maiste) [reply purse]: I resolve this conf as it is updated in https://github.com/TheLortex/githereum/pull/18#discussion_r866932817

File: purse (hunk 3 above)
Comment (maiste) [reply purse]: @CraigFe WDYT of this updated version with conditional?

**Review by maiste — APPROVED** [review]

**Review by TheLortex — APPROVED** [review]
Lgtm

**Comment by TheLortex** [comment]
IMG_20220519_100447

Proof of bus

**Comment by TheLortex** [comment]
IMG_20220519_100447

**Pull request merged**